source/deploy/postgres-migration.rst

```diff
@@ -206,7 +249,8 @@ Once we set the schema to a desired state, we can start migrating the **data** b
 
    WITH data only,
         workers = 8, concurrency = 1,
-        multiple readers per thread, rows per range = 50000,
+        multiple readers per thread, rows per range = 10000,
+        prefetch rows = 10000,
```

Comment:
@agnivade I'm inclined to reduce these numbers to stay on the safe side. WDYT?

Reply:
I think 10K should be safe. Maybe worth adding a comment, but I'd be comfortable with 10K default.

Reply:
> When using multiple readers per thread pgloader may be able to divide the reading work into several threads, as many as the concurrency setting, which needs to be greater than 1 for this option to kick be activated.



@isacikgoz As per the docs, it seems we don't even need these 2 options `multiple readers per thread, rows per range = 10000,`

Also, do we have any rough estimate how much memory would it take with 10k rows and 8 workers?

As it seems, some people run into heap exhaustion issue with these numbers.

Reply:
Hey @vish9812 I didn't get why `multiple readers per thread, rows per range = 10000` are not needed. It's to define how many rows should be read per thread. Do you mean the `prefetch rows` option?

For your question, we can't estimate that because there are some unbounded tables such as `FileInfo`.

Reply:
Also, we should add these 2 options as well: `batch rows` and `batch size`. And reduce the default value of batch rows, since current default 25k is more than our prefetch rows 10k.
As per docs default value, we should set the batch rows to 2500 for our 10k prefetch rows.

Reply:
@isacikgoz Because we have set the concurrency to 1. And docs seems to suggest that `multiple readers per thread` only kick in when `concurrency` > 1.

Reply:
@vish9812 Oh, I see what you mean now. I think we should add a note about that, because it should be better to show how to increase the throughput. Right now these values are there as a baseline. I just wanted to prevent someone from directly copying these values and going OOM.

Overall I agree that we can add either one of `batch size` or `batch rows` options. I think the safest option is `batch size` as it would mean `concurrency * workers * batch size` would be the max memory that `pgloader` supposed to consume. WDYT

Reply:
I think the correct max memory would be `workers * prefetch rows`.
And yes to show all the options, it would be good to highlight the `concurrency` and the `rows per range` options.

Since, we are using `prefetch rows` as number it would be easier to understand the batch as a number too. The docs default values uses 1/4th rows. So, we can use the `prefetch rows` to be 10k and `batch rows` to be 2500.

Reply:
Okay so there is a confusion here. Then why there is a need for `prefetch rows` value? And why `1/4` is the magic number there (is it maybe the default `workers` value is `4`)? I assume setting either `prefetch rows` or `batch size` should've been fine.

Reply:
Discussed offline, we are going to set `batch rows` to `2500` to have a reduced size of batching to let `pgloader` fail early if there is an issue with writes. The `batch rows` option seems like it doesn't have a memory impact as `prefetch rows` though but we are following the pattern in the default values.  Thanks for looking into this @vish9812